### DISCUSSION TITLE
To change the clock frequency  of STM32F103C

### DISCUSSION THREAD

**Discussion by gopalreddylakki** [stm32duino/Arduino_Core_STM32#2276] [General]
inSTM32F103C how we can change the clock frequency 

we need STM32F103C to be run at low power, we use a program that runs only 48mhz, in the 48mhz current consumption is 30 mA. My application only has 15 mA power only. 

i need the 8Mhz frequency to run, 8Mhz current is 15 mA

in this reason, i need to change the clock frequency from 48Mhz to 8Mhz

**Comment by fpistm** [comment]
https://github.com/stm32duino/Arduino_Core_STM32/wiki/Custom-definitions#systemclock_config

**Reply by gopalreddylakki** [reply]
Ok

if Can possible ,please  give an example 
extern "C" void SystemClock_Config() {
  // new clock config
}

**Reply by fpistm** [reply]
You can use STM32CubeMx to generate it.

**Comment by gopalreddylakki** [comment]
Thank you for helping i got it.
